src/industry_cmd.cpp

```diff
@@ -182,6 +182,11 @@ Industry::~Industry()
 
 	DeleteSubsidyWith(ST_INDUSTRY, this->index);
 	CargoPacket::InvalidateAllFrom(ST_INDUSTRY, this->index);
+
+	for (auto it = this->stations_near.Begin(); it != this->stations_near.End(); ++it) {
```

Comment:
would something like `for (Station *st: this->stations_near) {` work here?

Reply:
No, the iterator methods are incorrectly named for this. May be worth a separate PR :-)